enhancements/network/smart-nic-ovn-offload.md

```diff
@@ -0,0 +1,388 @@
+- [ ] Operational readiness criteria is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Summary
+
+Open vSwitch (OVS) Hardware Offload with NVIDIA ConnectX-5 improves OVS
+dataplane performance by offloading processing of OVS OpenFlow flows to
+hardware NIC. By integrating NVIDIA BlueField2 SmartNIC support in OpenShift,
+additional capabilities are added to accelerate and secure the OVN control
+plane by offloading OVS and OVN to the SmartNIC ARM processors.
+
+## Terminology
+
+- SmartNIC: A combined Network Interface Card (NIC) and CPU hardware which
+offloads processing tasks that the system CPU would normally handle
+- SmartNIC node(s): Baremetal servers where SmartNICs are plugged to
+- SmartNIC ARM cores: The embedded aarch64 system running on SmartNIC CPU(s)
+- SR-IOV: Single Root I/O Virtualization, a NIC technology that allows
+isolation of PCIe resources
+- PF: Physical Function, the physical ethernet controller that supports SR-IOV
+- VF: Virtual Function, the virtual PCIe device created from PF
+- VF representor: The port representor of the VF. VF and VF representor are
+created in pairs where VF appears in SmartNIC node and VF representor appears
+in SmartNIC ARM cores
+
+## Motivation
+
+Encryption, Encapsulation, and Virtual switching (IPsec, Geneve, VXLAN, OVS,
+Connection Tracking) to and from containers or VMs are basic building blocks
+of the Hybrid cloud and particularly of OCP. These functions are very compute
+intensive and can consume up to 40 cores from the server’s motherboard CPU
+for terminating 25Gigs of traffic. Terminating 100Gs without hardware offload
+is not feasible at all. Most of our customers are demanding all of the above
+without much use of the CPU on the motherboard of the server.
+
+### Goals
+
+- Only support NVIDIA BlueField2 because other implementations of SmartNIC are
+unknown
+- OVN-Kubernetes control plane including OVS and OVN controller are offloaded
+to SmartNIC, other CNIs are not currently planned
+- SR-IOV pods run on SmartNIC node using VF as pod default route interfaces
+- Manage SmartNIC node offload configurations in OpenShift Operators
+
+### Non-Goals
+
+- Support multiple SmartNICs on one OpenShift baremetal worker node
+- Support SmartNICs on OpenShift baremetal master nodes
+
+## Proposal
+
+- User installs SR-IOV Operator (OLM managed)
+- User creates one or more custom MachineConfigPools whose nodes are installed
+with SmartNIC hardware
+- User specifies NodeSelector labels in SR-IOV Network Operator(SNO) CRD to
+match on the custom MachineConfigPools
+- SR-IOV Network Operator renders and attaches SmartNIC MachineConfigs to the
+selected custom MachineConfigPools
+- SR-IOV Network Operator configures SmartNIC devices and exposes SR-IOV VFs
+as extended kubernetes resources
+- MachineConfigOperator reboots nodes in custom MachineConfigPool while
```

Comment:
Shall this step be put ahead of the above one?

Reply:
Yes, updated.